### ISSUE TITLE
Enable producers to see / edit their items in orders placed through 

### ISSUE THREAD

**Issue by jaycmb** [openfoodfoundation/wishlist#85]
## What is the need / problem?
Some Hubs or Farmers Markets want a more ‘hands off’ approach to the orders and transactions i.e. they want the customer to be able to place one order and pay once, but order adjustments to be handled by farmers rather than centrally by a Hub

Currently when an Order is placed the Producer can see their items in reports, but they cannot see the Order itself in the orders or BOM pages, and they cannot edit it i.e. remove items that they aren’t able to fulfil

This was a design decision in the early days when we were working on conversion of a single user ecommerce platform to multi-user. It was the same rationale as not allowing the Producer to see the Customer names on reports, which it is now clear does not meet the needs of many of our users (and is currently getting sorted)

This is also related to Split Payments (currently considering experiments with manual processing through Stripe to clarify the rules). If we do an interim split payments offering by handling directly through Stripe Connect, we only want to do the transfers once all the order information is correct.

If the Hub’s are ‘hands off’ it is better for the Producers to be able to amend orders that they haven’t been able to fulfil themselves

## Which type of users does this problem affect (and how many, if known)?

_- Hub Managers (MCFEs): Hub / Market managers who don’t want to handle order adjustments, refunds or payments to farmers.
_- Producers_

## Potential solutions that will solve the problem?

Remembering that we made all these choices and set up the permissions for who could see and do what on each page - products, reports, order cycles etc - it occurred to me that there is no real reason why we can’t just change the rules on the Orders page and the order editing

So the new rules for viewing and editing orders would be similar to order cycles. They would likely follow the new authorisation from the Hub to allow Producers to see Customer names or require creation of a new permission i.e. Hub allows Producer to manage orders

If a Hub authorises Producers to manage orders, then as a Producer

-   If I have products in an Order then i can see this order on my Orders page
-   If I have products in an Order then i can see this order on my Bulk Order Management page
-   I can remove my Products from a Customer’s order
-   I can edit quantities of my Products from a Customer’s order (both decrease and increase)
-   I can add my variants to a Customer’s order
-   Can I edit Adjustments?
-   Can I issue refunds? I think not . . I think the Producers could edit the orders and then the Hub (or instance service) does the refunds all in one go . . . although if have transferred the money already to the Producer’s stripe account this could hypothetically work

## Connected wishlist and discovery discussions* [list precedent discussions]

Migrated from Discourse: https://community.openfoodnetwork.org/t/enable-producers-to-see-edit-their-items-in-orders-placed-through-hub/1919

## Additional context

Originally opened
When: Apr '20
By:Kirsten

## Is Papercut or feature candidate?

Feature Candidate

**Comment by audez** [comment]
Yes we often have questions from producers who don't understand why orders don't appear in the orders list. They ask for them to be available in the Orders page. 
Could a first step be to simply display the orders in the Orders page? We could deactivate the "edit" button. 
At least the producers would see the orders, and don't have to go to the reports page..
Maybe It would be a papercut then?

**Comment by RachL** [comment]
@audez there is a spike that was selected as UK papercut. Once the spike is through, we will be able to know how to move forward https://github.com/openfoodfoundation/openfoodnetwork/issues/9049

Just displaying is not a papercut as you need to introduce the test to be sure the producer can't edit. Also you need to display _only_ products of this producer within the order and therefore perhaps there are many info we need to hide from the order (customer info if hub didn't allow them, order total, payment info etc). Perhaps the order page is not the best place for that?
That's why we have a spike first to understand how to scope this work.

**Comment by audez** [comment]
ooh ok i didn't see the spike, thanks!

**Comment by tschumilas** [comment]
fyi -  it seems this is related to another issue which is designated for a spike -  beyond just seeing their orders, 
  should a supplier to a hub be able to edit their orders?    We have producer/suppliers here who sell items weighed out at time of delivery.  So each time,  these producers need to inform the hub manager of the correct weight being delivered so the hub manager can adjust the product's weight.   Producers would like to be able to do this directly on theorder when they cut the product.  https://github.com/openfoodfoundation/wishlist/issues/172
